mypy-strict.ini

```diff
@@ -0,0 +1,33 @@
+# This is a MyPy config file but unlike mypy.ini, it enforces very strict typing
+# rules.  The intention is for this config file to be used to ENFORCE that
+# people are using mypy on codegen files.
+#
+# For now, only code_template.py is covered this way
+
+[mypy]
+python_version = 3.6
+
+strict_optional = True
+show_column_numbers = True
+warn_no_return = True
+disallow_any_unimported = True
+
+# Across versions of mypy, the flags toggled by --strict vary.  To ensure
+# we have reproducible type check, we instead manualy specify the flags
+warn_unused_configs = True
+disallow_any_generics = True
+disallow_subclassing_any = True
+disallow_untyped_calls = True
+disallow_untyped_defs = True
+disallow_incomplete_defs = True
+check_untyped_defs = True
+disallow_untyped_decorators = True
+no_implicit_optional = True
+warn_redundant_casts = True
+warn_unused_ignores = True
+warn_return_any = True
+no_implicit_reexport = True
```

Comment:
nit: the documented `.ini` option is `implicit_reexport`, so changing to `implicit_reexport = False` may be nice